FinalEngine.Rendering.OpenGL/OpenGLGPUResourceFactory.cs

```
@@ -78,4 +78,10 @@ public IVertexBuffer CreateVertexBuffer<T>(BufferUsageType type, IReadOnlyCollec
         ArgumentNullException.ThrowIfNull(data, nameof(data));
         return new OpenGLVertexBuffer<T>(this.invoker, this.mapper, this.mapper.Forward<BufferUsageHint>(type), data, sizeInBytes, stride);
     }
+
+    public IFrameBuffer CreateFrameBuffer(IReadOnlyList<ITexture2D> colorTargets, ITexture2D? depthTarget)
+    {
+        ArgumentNullException.ThrowIfNull(colorTargets, nameof(colorTargets));
+        return new OpenGLFrameBuffer(this.invoker, this.mapper, colorTargets, depthTarget);
```

Before we create the new frame buffer, let's use the pipeline here to ensure that `colorTargets` does not exceed the maximum color targets using the new property in `IPipeline`. This ensures the check is made for each frame buffer that is created and it's not forgotten.

---

If it exceeds the maximum amount, we should throw an exception - maybe call it `FrameBufferTargetException` or something similar.

The reason we should throw an exception here is to ensure that the user doesn't create a frame buffer and assume they are drawing to all color targets they specify (this is the case for your current approach).